### ISSUE TITLE
failed to complete precursor corrections for Impact II data

### ISSUE THREAD

**Issue by annapopova-scripps** [elnurgar/mzxml-precursor-corrector#4]
Dear developer,

I am extremely interested in the software you are distributing.  Precursor m/z correction was a long time issue for LC-MSMS data I  normally acquire on Bruker Impact II instrument.  These are negative ionization mode, profile,  regular DDA type of data. Following acquisition, I used MSconvert to transform  a subset of .d to either mzML or mzXML. format The program recognizes the format , continues scanning the file provided, than briefly exits, likely w/o finding any MS1 precursors.
I am wondering if I could set a subset of my  mzML data for troubleshooting. Would negative mode be a burden? These are oligonucleotide, not proteomics samples.

Thank you in advance.

**Comment by elnurgar** [comment]
Dear colleague,

Can you please send me an example of your raw file to elnurgar@gmail.com? I used this script on our Bruker Impact II without any problems.

**Comment by annapopova-scripps** [comment]
Thank you for agreeing to help!
 I shared the dropbox folder  with raw and mzML converted data.
https://www.dropbox.com/scl/fo/9uk89sq9a0f4zjdyav52z/AIJRuI-pZgO5l79xMwYqNxk?rlkey=mms8bil989wtzwaobm3hdg6xx&st=jvnpq9pd&dl=0
Anna

________________________________

**Comment by elnurgar** [comment]
I think there is an issue with file conversion. The mzml file is very huge (27 Gb) comparing to raw data (1 Gb). I have used msconvert with these parameters to generate a 389 Mb mzml file:
![image](https://github.com/user-attachments/assets/0f89ca4a-e615-4df1-abbe-d94af0563ecb)

After that you can use this script to corrige precursor values. 

Hope this will help you. 

Elnur

**Comment by annapopova-scripps** [comment]
Hi,

I appreciate your feedback on this.  I did not do pick picking to mzml convert. I guess, I missed that step in the instructions.
I will try and write you back.

Anna
________________________________

**Comment by annapopova-scripps** [comment]
Hi again,

I repeated mzML conversion few more times with  1) centroided MS2 only or 2)  centroided MS1 and MS2 data with no outcome.
Will be glad to get your input on the problem.

Best,
Anna


________________________________

**Comment by elnurgar** [comment]
You convert using parameters peak picking ms level 1-2 and mzml 64 bit and then you place your mzml file in the same folder with script exe file? Can you send me your new mzml file?

**Comment by annapopova-scripps** [comment]
Thanks,

I just added a new version to the Dropbox. It is a subset of .d between 30-35 min to minimize processing time.

Anna

**Comment by elnurgar** [comment]
I have just checked with your file. Actually, I cannot use this file as the first spectra in the file should be MS1 spectra and your file begins with ms2 spectra. I tried to convert the raw file with the parameters of msconvert:
![image](https://github.com/user-attachments/assets/c99a531a-b8a7-410c-b104-5da71fafd31c)


Then, I put the exe file in the same folder as the mzml file and launch the exe file which will generate new mzml with precursor values corrected:
![image](https://github.com/user-attachments/assets/06a706e1-3d7e-453e-88a4-4f2783498961)

It works without any problems.
